doc/sfp-cmis/cmis.md

```diff
@@ -0,0 +1,308 @@
+by a remote user to triage either system-side or line-side issues and help isolate failing hardware.
+
+## 2.2 Functional Description
+
+- The **pmon#xcvrd** will detect the module type of the attched transceivers and post the
+information to the **STATE_DB**
+- If the transceiver is a QSFPDD CMIS module, the **pmon#xcvrd** will activate the desired
+application mode as per the current port mode.
+- If the transceiver is a QSFPDD CMIS module, the **show interfaces transceiver eeprom**
+should display the advertised applications of the transceiver.
+- If the transceiver is a QSFPDD CMIS module, the users will be able to enable or disable the
+loopbacks at the transceiver layer to triage link issues.
+- If the transceiver is a QSFPDD CMIS module, the users will be able to check the state of
+the application initialization.
+
+# Design
+
+## 3.1 Overview
+
+![](images/002.png)
+
+### 3.1.1 sonic-platform-daemons
+
+#### 3.1.1.1 sonic-xcvrd (modified)
+
+The transceiver daemon will be enhanced as below:
+
+- Post the QSFP-DD infomration to STATE_DB
+- Add a new CMIS task to handle the multiple CMIS initializations in parallel
+
+### 3.1.2 sonic-platform-common
+
+#### 3.1.2.1 sonic_platform_base/sonic_sfp/inf8628.py (modified)
+
+This library will be enhanced as below
+
+- Add support for decoding the advertised applications of the QSFPDD CMIS transceiver
+- Add support for decoding the DOM information of the QSFPDD CMIS transceiver
+
+#### 3.1.2.2 sonic_platform_base/sfp_standard.py (new)
+
+This is a new common SFP base class that's supposed to be inherited by the SFP drivers
+of the individual platforms, and it will later be refactored when **sfp-refactor** is
+open for business.
+
+### 3.1.3 sonic-utilities
+
+#### 3.1.3.1 sfputil (modified)
+
+This utility will be enhanced to support the following features
+
+- Show the current application mode of the QSFPDD CMIS transceiver
+- Show the current configuration errors of the QSFPDD CMIS transceiver
+- Show the loopback capabilities of the QSFPDD CMIS transceiver
+- Enable/Disable the loopback controls provided by the QSFPDD CMIS transceiver
+
+## 3.2 DB Changes
+
+### 3.2.1 CONFIG_DB
+
+No modifications made.
+
+### 3.2.2 APPL_DB
+
+No modifications made.
+
+### 3.2.3 STATE_DB
+
+The **TRANSCEIVER_INFO** table will be extended as below if the transceiver is compliant with CMIS 4.0 onward:
+
+```
+  "TRANSCEIVER_INFO|Ethernet0": {
+    "type": "hash",
+    "value": {
+      "application_advertisement": "{1: {'host_if': '400GAUI-8 C2M (Annex 120E)', 'media_if': '400GBASE-DR4 (Cl 124)'}, 2: {'host_if': '100GAUI-2 C2M (Annex 135G)', 'media_if': '100G-FR/100GBASE-FR1 (Cl 140)'}}",
+      "cable_length": "0.0",
+      "cable_type": "Length SMF(km)",
+      "connector": "SN (Mini CS)",
+      "connector_type": "SN (Mini CS)",
+      "encoding": "N/A",
+      "ext_identifier": "N/A",
+      "ext_rateselect_compliance": "N/A",
+      "hardware_rev": "01",
+      "lane_count": "4",
```

Comment:
this is media lane count, right? (host lane count would be 8)

Reply:
yep, this is the effective lane counts in the line side